libtransmission/handshake.cc

```diff
@@ -771,67 +769,64 @@ ReadState tr_handshake::can_read(tr_peerIo* peer_io, void* vhandshake, size_t* p
         }
         else if (handshake->is_state(State::AwaitingPadC))
         {
-            ready_for_more = peer_io->readBufferSize() >= handshake->pad_c_len_;
+            ready_for_more = peer_io->read_buffer_size() >= handshake->pad_c_len_;
         }
         else if (handshake->is_state(State::AwaitingPadD))
         {
-            ready_for_more = peer_io->readBufferSize() >= handshake->pad_d_len_;
+            ready_for_more = peer_io->read_buffer_size() >= handshake->pad_d_len_;
         }
         else if (handshake->is_state(State::AwaitingIa))
         {
-            ready_for_more = peer_io->readBufferSize() >= handshake->ia_len_;
+            ready_for_more = peer_io->read_buffer_size() >= handshake->ia_len_;
         }
     }
 
     return ret;
 }
 
-void tr_handshake::on_error(tr_peerIo* io, short what, void* vhandshake)
+void tr_handshake::on_error(tr_peerIo* io, tr_error const& error, void* vhandshake)
 {
-    auto const errcode = errno;
     auto* handshake = static_cast<tr_handshake*>(vhandshake);
 
-    if (io->socket.is_utp() && !io->isIncoming() && handshake->is_state(State::AwaitingYb))
+    if (io->is_utp() && !io->is_incoming() && handshake->is_state(State::AwaitingYb))
     {
         // the peer probably doesn't speak µTP.
 
-        auto const info_hash = io->torrentHash();
+        auto const info_hash = io->torrent_hash();
         auto const info = handshake->mediator_->torrent(info_hash);
 
         /* Don't mark a peer as non-µTP unless it's really a connect failure. */
-        if ((errcode == ETIMEDOUT || errcode == ECONNREFUSED) && info)
+        if ((error.code == ETIMEDOUT || error.code == ECONNREFUSED) && info)
         {
             handshake->mediator_->set_utp_failed(info_hash, io->address());
         }
 
-        if (handshake->mediator_->allows_tcp() && io->reconnect() == 0)
+        if (handshake->mediator_->allows_tcp() && io->reconnect())
```

Comment:
Is that an intended change (aka, some bug fix)?
Previously we were doing `== 0`, and now it's the same as `!= 0`.

Reply:
Yes, https://github.com/transmission/transmission/pull/4372/commits/0fed1f43436db1981cf8467261498adc5407255a -- reconnect now returns a bool. Previously it used the pre-bool-C-ism of returning 0 on success, -1 on error.